File: src/parser/packet/ReceiveData.java
package com.yuhki50.lpr9201.parser.packet;

import java.util.Arrays;

/**
 * 受信データ
 */
public class ReceiveData implements IParsePacket {
    /**
     * 結果コード
     */
    protected static final int RESULT_CODE = 0x83;

    /**
     * データ長のバイト数
     */
    protected static final int DATA_LENGTH_BYTE_SIZE = 2;

    /**
     * 受信したリザルトデータ
     */
    protected Result result;

    /**
     * 受信データ
     *
     * @param result 結果クラス
     */
    public ReceiveData(Result result) {
        this.result = result;
    }

    /**
     * 結果コードを取得する
     *
     * @return 結果コード
     */
    public int getResultCode() {
        return RESULT_CODE;
    }

    /**
     * データ長のバイト数を取得する
     *
     * @return データ長のバイト数
     */
    public int getDataLengthByteSize() {
        return DATA_LENGTH_BYTE_SIZE;
    }

    /**
     * パース可能か
     *
     * @return true:パース可能, false: パース不可
     */
    public boolean canParse() {
        return this.result != null && this.result.resultCode == RESULT_CODE;
    }

    /**
     * データを取得する
     *
     * @return データ
     */
    public int[] getDatas() {
        int[] datas = this.result.datas;
        return Arrays.copyOf(datas, datas.length);
    }
}

File: src/parser/packet/ReceiveData.ts
/// <reference path='IParsePacket.ts' />
/// <reference path='Result.ts' />

/**
 * 受信データ
 */
class ReceiveData implements IParsePacket {
    /**
     * 結果コード
     */
    //protected static const RESULT_CODE : number = 0x83;
    public static RESULT_CODE : number = 0x83;

    /**
     * データ長のバイト数
     */
    //protected static const DATA_LENGTH_BYTE_SIZE : number = 2;
    public static DATA_LENGTH_BYTE_SIZE : number = 2;

    /**
     * 受信したリザルトデータ
     */
    protected result : Result;

    /**
     * 受信データ
     *
     * @param result 結果クラス
     */
    public constructor(result : Result) {
        this.result = result;
    }

    /**
     * 結果コードを取得する
     *
     * @return 結果コード
     */
    public getResultCode() : number {
        return ReceiveData.RESULT_CODE;
    }

    /**
     * データ長のバイト数を取得する
     *
     * @return データ長のバイト数
     */
    public getDataLengthByteSize() : number {
        return ReceiveData.DATA_LENGTH_BYTE_SIZE;
    }

    /**
     * パース可能か
     *
     * @return true:パース可能, false: パース不可
     */
    public canParse() : boolean {
        return this.result != null && this.result.resultCode == ReceiveData.RESULT_CODE;
    }

    /**
     * データを取得する
     *
     * @return データ
     */
    public getDatas() : number[] {
        //var datas = this.result.datas;
        //return Arrays.copyOf(datas, datas.length);
        return this.result.datas.slice(0);
    }
}
